<a href="https://colab.research.google.com/github/stephintoto/data-aquisition/blob/main/data_aqua.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

This dataset provides data on the movies and TV shows available on Netflix, a popular streaming service. we can use this to discover the movies and TV shows available on Netflix, as well as their title, type, director, cast, country, date added, release year, rating, duration, genre and description.


In [6]:
import pandas as pd


In [7]:
df=pd.read_csv('NetFlix.csv')
df.head()

show_id     type                                      title        director  \
0      s1  TV Show                                         3%             NaN   
1     s10    Movie                                       1920    Vikram Bhatt   
2    s100    Movie                                 3 Heroines  Iman Brotoseno   
3   s1000    Movie  Blue Mountain State: The Rise of Thadland    Lev L. Spiro   
4   s1001  TV Show                             Blue Planet II             NaN   

                                                cast         country  \
0  João Miguel, Bianca Comparato, Michel Gomes, R...          Brazil   
1  Rajneesh Duggal, Adah Sharma, Indraneil Sengup...           India   
2  Reza Rahadian, Bunga Citra Lestari, Tara Basro...       Indonesia   
3  Alan Ritchson, Darin Brooks, James Cade, Rob R...   United States   
4                                 David Attenborough  United Kingdom   

  date_added  release_year rating  duration  \
0  14-Aug-20          2020  TV-MA         4   
1  15-Dec-17          2008  TV-MA       143   
2   5-Jan-19          2016  TV-PG       124   
3   1-Mar-16          2016      R        90   
4   3-Dec-18          2017   TV-G         1   

                                              genres  \
0  International TV Shows, TV Dramas, TV Sci-Fi &...   
1     Horror Movies, International Movies, Thrillers   
2        Dramas, International Movies, Sports Movies   
3                                           Comedies   
4  British TV Shows, Docuseries, Science & Nature TV   

                                         description  
0  In a future where the elite inhabit an island ...  
1  An architect and his wife move into a castle t...  
2  Three Indonesian women break records by becomi...  
3  New NFL star Thad buys his old teammates' belo...  
4  This sequel to the award-winning nature series...

summary statistics of dataset :  display the summary statistics for all numerical columns in the dataset, including count, mean, standard deviation, minimum, 25th percentile, median (50th percentile), 75th percentile, and maximum values.





In [ ]:
summary=df.describe()
summary

release_year     duration
count   7787.000000  7787.000000
mean    2013.932580    69.122769
std        8.757395    50.950743
min     1925.000000     1.000000
25%     2013.000000     2.000000
50%     2017.000000    88.000000
75%     2018.000000   106.000000
max     2021.000000   312.000000

inorder to get the size ie no. of rows and columns we use the shape attribute
df.shape[0] gives the number of rows
df.shape[1] gives the number of columns

In [ ]:

dataset_size = df.shape
print(f"Number of rows: {dataset_size[0]}")
print(f"Number of columns: {dataset_size[1]}")


Number of rows: 7787
Number of columns: 12


to get the data type of each column

In [ ]:
data_type=df.dtypes
data_type

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration         int64
genres          object
description     object
dtype: object

df.isnull().sum() calculates the number of missing values (NaN) in each column
missing_values[missing_values > 0] filters out columns that have no missing values, showing only those with missing data.

In [ ]:
missing_values = df.isnull().sum()
columns_missing_values = missing_values[missing_values > 0]
print(columns_missing_values)

director      2389
cast           718
country        507
date_added      10
rating           7
dtype: int64


df[categorical_column].unique() returns an array of the unique values in the specified column
df[categorical_column].nunique() returns the number of unique values in the column

In [11]:
categorical_column = 'type'
unique_values = df[categorical_column].unique()
unique_count = df[categorical_column].nunique()
print(f"Unique values in '{categorical_column}': {unique_values}")
print(f"Number of unique values: {unique_count}")


Unique values in 'type': ['TV Show' 'Movie']
Number of unique values: 2


df['duration'].mean() calculates the average duration
df['duration'].median() calculates the middle value
df['duration'].std() calculates the standard deviatio

In [ ]:
numerical_column = 'duration'
mean_value = df[numerical_column].mean()
median_value = df[numerical_column].median()
std_deviation = df[numerical_column].std()

# Print the results
print(f"Mean of '{numerical_column}': {mean_value}")
print(f"Median of '{numerical_column}': {median_value}")
print(f"Standard Deviation of '{numerical_column}': {std_deviation}")


Mean of 'duration': 69.12276871709258
Median of 'duration': 88.0
Standard Deviation of 'duration': 50.950743343471515


Q1 and Q3: These are the 25th and 75th percentiles, represents the lower and upper bounds of the middle 50% of the data,
IQR: The range within which the central 50% of the data lies,
Lower and Upper Bounds: Any values below or above these bounds are considered outliers.

In [8]:

num_col = 'duration'
Q1 = df[num_col].quantile(0.25)
Q3 = df[num_col].quantile(0.75)
IQR = Q3 - Q1

lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR


outliers = df[(df[num_col] < lower_bound) | (df[num_col] > upper_bound)]

print(f"Lower bound: {lower_bound}")
print(f"Upper bound: {upper_bound}")
print(f"Number of outliers in '{num_col}': {outliers.shape[0]}")
print(outliers[[num_col]])


Lower bound: -154.0
Upper bound: 262.0
Number of outliers in 'duration': 1
      duration
7741       312


.corr() method to find the correlation coefficient between the two numerical columns.
Positive Correlation: If the correlation is close to 1, it suggests that newer shows/movies tend to have longer or short durations.
Negative Correlation: If the correlation is close to -1, it suggests that older shows/movies tend to have longer or shorter duration.
No Correlation: If the correlation is close to 0, it suggests that there is no linear relationship between the release year and duration.

here it is close to 0 so there is no realation

In [9]:

correlation = df['release_year'].corr(df['duration'])

print(f"Correlation between 'release_year' and 'duration': {correlation}")


Correlation between 'release_year' and 'duration': -0.24083598848535753


groupby() method to group the DataFrame by the country column.

.agg() method to calculate various statistics like mean, sum, min, max, count for the duration column.

shows the agg stat of each country

In [10]:

aggregated_stats = df.groupby('country')['duration'].agg(['mean', 'sum', 'min', 'max', 'count', 'std', 'median'])

print(aggregated_stats)


                                                      mean   sum  min  max  \
country                                                                      
Argentina                                            59.56  2978    1  116   
Argentina, Brazil, France, Poland, Germany, Den...   96.00    96   96   96   
Argentina, Chile                                     78.00    78   78   78   
Argentina, Chile, Peru                              100.00   100  100  100   
Argentina, France                                    87.00    87   87   87   
...                                                    ...   ...  ...  ...   
Venezuela                                           119.00   119  119  119   
Venezuela, Colombia                                  82.00    82   82   82   
Vietnam                                             102.00   510   92  116   
West Germany                                        150.00   150  150  150   
Zimbabwe                                            100.00   100

**SUMMARY**


The dataset contains 7,787 rows and 12 columns, including both categorical and numerical data.

Missing values were identified in several columns:
director: 2,389 missing values
cast: 718 missing values
country: 507 missing values
date_added: 10 missing values
rating: 7 missing values

 The dataset shows a wide variety of content in terms of duration and production across different countries.

 missing data in key columns like director and cast may require attention

there is no strong correlation between the release year and duration, suggesting that content length has remained relatively stable over time.